test/framework/shootmigrationtest.go

```diff
@@ -0,0 +1,227 @@
+		nodeNames[i] = node.Name
+	}
+	sort.Strings(nodeNames)
+	return
+}
+
+// GetMachineDetails uses the seedClient to fetch all Machine names and the names of their corresponding Nodes
+func (t *ShootMigrationTest) GetMachineDetails(ctx context.Context, seedClient kubernetes.Interface) (machineNames, machineNodes []string, err error) {
+	machineList := unstructured.UnstructuredList{}
+	machineList.SetAPIVersion("machine.sapcloud.io/v1alpha1")
+	machineList.SetKind("Machine")
+	t.GardenerFramework.Logger.Infof("Getting machine details in namespace: %s", t.SeedShootNamespace)
+	if err := seedClient.Client().List(ctx, &machineList, client.InNamespace(t.SeedShootNamespace)); err != nil {
+		t.GardenerFramework.Logger.Errorf("Error while getting Machine details, %s", err.Error())
+		return nil, nil, err
+	}
+
+	t.GardenerFramework.Logger.Infof("Found: %d Machine items", len(machineList.Items))
+
+	machineNames = make([]string, len(machineList.Items))
+	machineNodes = make([]string, len(machineList.Items))
+	for i, machine := range machineList.Items {
+		t.GardenerFramework.Logger.Infof("%d. Machine Name: %s, Node Name: %s", i, machine.GetName(), machine.GetLabels()["node"])
+		machineNames[i] = machine.GetName()
+		machineNodes[i] = machine.GetLabels()["node"]
+	}
+	sort.Strings(machineNames)
+	sort.Strings(machineNodes)
+	return
+}
+
+// PopulateBeforeMigrationComparisonElements fills the ShootMigrationTest.ComparisonElementsBeforeMigration with the necessary Machine details and Node names
+func (t *ShootMigrationTest) PopulateBeforeMigrationComparisonElements(ctx context.Context) (err error) {
+	t.ComparisonElementsBeforeMigration.MachineNames, t.ComparisonElementsBeforeMigration.MachineNodes, err = t.GetMachineDetails(ctx, t.SourceSeedClient)
+	if err != nil {
+		return
+	}
+	t.ComparisonElementsBeforeMigration.NodeNames, err = t.GetNodeNames(ctx, t.ShootClient)
+	return
+}
+
+// PopulateAfterMigrationComparisonElements fills the ShootMigrationTest.ComparisonElementsAfterMigration with the necessary Machine details and Node names
+func (t *ShootMigrationTest) PopulateAfterMigrationComparisonElements(ctx context.Context) (err error) {
+	t.ComparisonElementsAfterMigration.MachineNames, t.ComparisonElementsAfterMigration.MachineNodes, err = t.GetMachineDetails(ctx, t.TargetSeedClient)
+	if err != nil {
+		return
+	}
+	t.ComparisonElementsAfterMigration.NodeNames, err = t.GetNodeNames(ctx, t.ShootClient)
+	return
+}
+
+// CompareElementsAfterMigration compares the Machine details, Node names and Pod statuses before and after migration and returns error if there are diferences.
+func (t *ShootMigrationTest) CompareElementsAfterMigration() error {
+	if !reflect.DeepEqual(t.ComparisonElementsBeforeMigration.MachineNames, t.ComparisonElementsAfterMigration.MachineNames) {
+		return errors.Errorf("Initial Machines %s, does not match after-migrate Machines %s", t.ComparisonElementsBeforeMigration.MachineNames, t.ComparisonElementsAfterMigration.MachineNames)
```

Comment:
Ref https://github.com/golang/go/wiki/CodeReviewComments#error-strings

```suggestion
		return errors.Errorf("initial Machines %s do not match after-migrate Machines %s", t.ComparisonElementsBeforeMigration.MachineNames, t.ComparisonElementsAfterMigration.MachineNames)
```